MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -1188,6 +1183,76 @@ public string GetVideoQualityParam(EncodingJobInfo state, string videoEncoder, E
                     qmin,
                     qmax);
             }
+            else if (string.Equals(videoEncoder, "libvpx-vp9", StringComparison.OrdinalIgnoreCase)) // vp9
+            {
+                // When `-deadline` is set to `good` or `best`, `-cpu-used` ranges from 0-5.
+                // When `-deadline` is set to `realtime`, `-cpu-used` ranges from 0-15.
+                // Resources:
+                //   * https://trac.ffmpeg.org/wiki/Encode/VP9
+                //   * https://superuser.com/questions/1586934
+                //   * https://developers.google.com/media/vp9
+                switch (encodingOptions.EncoderPreset)
+                {
+                    case "veryslow":
+                        param += " -deadline best -cpu-used 0";
+                        break;
+
+                    case "slower":
+                        param += " -deadline best -cpu-used 2";
+                        break;
+
+                    case "slow":
+                        param += " -deadline best -cpu-used 3";
+                        break;
+
+                    case "medium":
+                        param += " -deadline good -cpu-used 0";
+                        break;
+
+                    case "fast":
+                        param += " -deadline good -cpu-used 1";
+                        break;
+
+                    case "faster":
+                        param += " -deadline good -cpu-used 2";
+                        break;
+
+                    case "veryfast":
+                        param += " -deadline good -cpu-used 3";
+                        break;
+
+                    case "superfast":
+                        param += " -deadline good -cpu-used 4";
+                        break;
+
+                    case "ultrafast":
+                        param += " -deadline good -cpu-used 5";
+                        break;
+
+                    default:
+                        param += " -deadline good -cpu-used 1";
+                        break;
+                }
+
+                // TODO: until VP9 gets its own CRF, base CRF on H.265.
+                int h265Crf = encodingOptions.H265Crf;
+                int defaultVp9Crf = 31;
+                if (h265Crf >= 0 && h265Crf <= 51)
+                {
+                    var vp9Crf = Convert.ToInt32(h265Crf * 1.12);
```

Comment:
Where does `1.12` come from?

Reply:
I added a comment in the code, but I'll reproduce it here for posterity.


> This conversion factor is chosen to match the default CRF for H.265 to the recommended 1080p CRF from Google. The factor also maps the logarithmic CRF scales of x265 [0, 51] to that of VP9 [0, 63] relatively 
>
> Resources:
>   * https://developers.google.com/media/vp9/settings/vod